plugins/Kaleidoscope-CharShift/src/kaleidoscope/plugin/CharShift.cpp

```diff
@@ -34,6 +34,10 @@ namespace plugin {
 // CharShift class variables
 CharShift::KeyPair const * CharShift::progmem_keypairs_{nullptr};
 uint8_t CharShift::num_keypairs_{0};
+CharShift::GetNumKeyPairsFunction CharShift::numKeyPairs_ =
+    CharShift::numProgmemKeyPairs;
```

Comment:
I feel like this is unnecessary, but I'm not sure I'm right.  Couldn't the same `num_keypairs_` variable be used, regardless of where the data is stored?  As long as we're reading values from either PROGMEM _or_ EEPROM, but not both (perhaps by using an index offset), there's no need to use three bytes to store a separate count, plus the function pointer, right?  Or are you thinking the user might want to switch back and forth between the two without power-cycling the keyboard?

Reply:
I'm thinking that a user might want to use both at the same time, store some pairs in PROGMEM, some in EEPROM. Not sure if that makes sense for CharShift - it does for Macros and TapDance, where I copied the basic architecture from.